dspy/predict/predict.py

```diff
@@ -12,6 +12,7 @@ def __init__(self, signature, **config):
         self.stage = random.randbytes(8).hex()
         self.signature = signature #.signature
         self.config = config
+        self.lm = None
```

Comment:
In the grand scheme of things this makes sense. We'll need good docs about the behavior of LMs of modules.

Reply:
Yes! And the codebase should be restructured to not use any with dsp.context.settings(lm=model) context managers. There should be mainly one way to keep track of LMs on modules (self.lm) and a canonical way of setting these (e.g. program.module1.to(lm)).

(We could also have a student_lm and teacher_lm object, so that you don't have to manually swap out teacher and student LMs during optimization for the teacher program. However, this is not that important.)